x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/EngineIndexService.java

```diff
@@ -177,13 +186,68 @@ public void getEngine(String resourceName, ActionListener<Engine> listener) {
         }));
     }
 
+    private static String getEngineAliasName(Engine engine) {
+        return ENGINE_ALIAS_PREFIX + engine.name();
+    }
+
     /**
      * Creates or updates the {@link Engine} in the underlying index.
      *
      * @param engine The engine object.
      * @param listener The action listener to invoke on response/failure.
      */
     public void putEngine(Engine engine, ActionListener<IndexResponse> listener) {
+        validateIndices(engine);
+
+        createOrUpdateAlias(engine, listener.delegateFailure((l, aliasResponse) -> { updateEngine(engine, l); }));
+
+    }
+
+    private void createOrUpdateAlias(Engine engine, ActionListener<AcknowledgedResponse> listener) {
+
+        final Metadata metadata = clusterService.state().metadata();
+        final String engineAliasName = getEngineAliasName(engine);
+
+        IndicesAliasesRequestBuilder requestBuilder = null;
+        if (metadata.hasAlias(engineAliasName)) {
```

Comment:
A few cases we need to check:
- The alias has not yet been created
- Alias is created, and we need to update the indices
- Alias is created, and there's nothing to do. We don't do an operation or validation would fail as there is no action.